foundry/foundry.py

```diff
@@ -140,210 +140,126 @@ def __init__(
+            metadatas = self.get_metadata_by_query(query, limit)
 
-        # if search returns multiple results, this automatically uses first result, while warning the user
-        if len(res) > 1:
-            warnings.warn("Multiple datasets found for the given search query. Using first dataset")
-        res = res[0]
+        if len(metadatas) == 0:
+            raise Exception(f"load: No results found for the query '{query}'")
 
-        try:
-            res["dataset"] = res["projects"][self.config.metadata_key]
-        except KeyError as e:
-            raise Exception(f"load: not able to index with metadata key {self.config.metadata_key}") from e
+        foundry_datasets = []
+        for metadata in metadatas:
+            ds = self.dataset_from_metadata(metadata)
+            if ds:
+                foundry_datasets.append(ds)
 
-        del res["projects"][self.config.metadata_key]
+        print(f"Search for '{query}' returned {len(foundry_datasets)} foundry datasets out of {len(metadatas)} matches")
+        return foundry_datasets
 
-        # TODO: Creating a new Foundry instance is a problematic way to update the metadata,
-        # we should find a way to abstract this.
+    def list(self, limit: int = None):
+        """List available Foundry datasets
 
-        self.dc = res['dc']
-        self.mdf = res['mdf']
-        self.dataset = FoundryDataset(**res['dataset'])
+        Args:
+            limit (int): maximum number of results to return
 
-        if download:  # Add check for package existence
-            self.download(
-                interval=kwargs.get("interval", 10), globus=globus, verbose=verbose
-            )
+        Returns
+            (pandas.DataFrame): DataFrame with summary list of Foundry datasets including name, title, publication year, and DOI
+        """
+        return self.search(limit=limit)
 
-        return self
+    def dataset_from_metadata(self, metadata: dict) -> FoundryDataset:
+        """ Converts the result of a forge query to a FoundryDatset object
 
-    def search(self, q=None, limit=None):
-        """Search available Foundry datasets
-        q (str): query string to match
-        limit (int): maximum number of results to return
+        Args:
+            metadata (dict): result from a forge query
 
-        Returns
-        -------
-            (pandas.DataFrame): DataFrame with summary list of Foundry data packages including name, title, publication year, and DOI
+        Returns:
+            FoundryDataset: a FoundryDatset object created from the metadata
         """
-        if not q:
-            q = None
-        res = (
-            self.forge_client.match_field(
-                "mdf.organizations", self.config.organization)
-            .match_resource_types("dataset")
-            .search(q, limit=limit)
-        )
+        try:
+            if 'project' in metadata.keys():
+                schema = FoundrySchema(**metadata['projects']['foundry'])
```

Comment:
That's fair - I removed this logic, as the intent was to isolate the exception as derived from the instantiation of the FoundryDataset object (in case we wanted to allow formation of partial objects) but that isn't really implemented or necessary. I've reduced the code so that any missing pieces of the metadata will trigger the exception.